src/controller/content-steering-controller.ts

```diff
@@ -92,6 +92,27 @@ export default class ContentSteeringController
     hls.off(Events.ERROR, this.onError, this);
   }
 
+  getPathwaysList(levels: Level[]) {
+    return levels.reduce((pathways, level) => {
+      if (pathways.indexOf(level.pathwayId) === -1) {
+        pathways.push(level.pathwayId);
+      }
+      return pathways;
+    }, [] as string[]);
+  }
```

Comment:
Make this a `pathways` getter (we should not expose a method that produces a list of pathways based on an arbitrary levels argument):

```suggestion
  get pathways() {
    return (this.levels || []).reduce((pathways, level) => {
      if (pathways.indexOf(level.pathwayId) === -1) {
        pathways.push(level.pathwayId);
      }
      return pathways;
    }, [] as string[]);
  }
```